Automattic/wp-calypso
# Hide features where no plans use and remove unavailable Woo Express features

Related to #84178

## Proposed Changes

* Remove the following items from Woo Express plans page:
    * `Customizable Product Kits` or `Assembled Products and kits`
    * `Referral Program`
    * `Customer Birthday Emails`
    * `Loyalty Points Program`
* Hide feature rows from plan page when no plan is using that particular feature
* Hide entire feature group from plan page when no feature is shown under the group

## Testing Instructions

**Testing unavailable Woo Express features**
1. Go to https://woo.com/start and go through NUX until creating a free trial site
1. Go to http://calypso.localhost:3000/plans/SITE_URL
1. Go to Upgrades > Plans page and expand the compare grid
1. Observe the features mentioned are not shown in the page
1. Go to http://calypso.localhost:3000/plans/my-plan/trial-expired/SITE_URL and expand the compare grid
1. Observe the features mentioned are not shown in the page

Removed items:

<img width="600" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/c92e34b1-91bd-4fee-9e18-0bfa69cb87aa">
<img width="600" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/2855e375-e1b2-471f-aeb0-d9d725dcc6ea">
<img width="600" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/5e36d518-81b6-45bd-a349-a247fee08156">


**Testing features and feature group removal**
1. In your terminal, run the following bash command to apply a [patch](https://gist.githubusercontent.com/ilyasfoo/1cf26a4b64fc9102a8929e440e82234d/raw/4e05e5c1bbe20ce5034f0dffa60eb1b0c6d348bd/changes.diff): `(curl -sL https://gist.githubusercontent.com/ilyasfoo/1cf26a4b64fc9102a8929e440e82234d/raw/4e05e5c1bbe20ce5034f0dffa60eb1b0c6d348bd/changes.diff; echo) | git apply`
1. Go to Upgrades > Plans page and expand the compare grid
3. Observe that the entire `Shipping` feature group is not shown
